fastrtps fails to compile on greater than or equal to msvc version 19.38.32926.95

### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

No error

### Current behavior

Hello, I'm a member of Microsoft [VCPKG](https://github.com/microsoft/vcpkg), when I built this port in an internal version of MSVC, it failed with following errors:
`E:\Jim\vcpkg\buildtrees\fastrtps\src\v2.11.1-d9ab4e5ea6\include\fastrtps/utils/TimedMutex.hpp(177): error C2065: '_Thrd_success': undeclared identifier`

### Steps to reproduce

Build with MSVC 19.38.32926.95 and above, _Thrd_success is modified to _Thrd_result::_Success since https://github.com/microsoft/STL/pull/3897, it's an upcoming release.

### Fast DDS version/commit

2.7.0. For details, see: [fastrtps/vcpkg.json](https://github.com/microsoft/vcpkg/blob/master/ports/fastrtps/vcpkg.json)

### Platform/Architecture

Other. Please specify in Additional context section.

### Transport layer

Default configuration, UDPv4 & SHM

### Additional context

Windows
Visual Studio
The STL developers suggest that we should not be using internal STL identifiers. Also practice STL developer if you have other questions.

### XML configuration file

_No response_

### Relevant log output

_No response_

### Network traffic capture

_No response_

--- a/include/fastrtps/utils/TimedMutex.hpp
+++ b/include/fastrtps/utils/TimedMutex.hpp
@@ -31,7 +31,7 @@
 extern int clock_gettime(
         int,
         struct timespec* tv);
-#endif
+#endif // if defined(_MSC_FULL_VER) && _MSC_FULL_VER >= 193632528
 
 #elif _GTHREAD_USE_MUTEX_TIMEDLOCK
 #include <mutex>
@@ -194,7 +194,7 @@ private:
 
     _Mtx_t mutex_;
 };
-#endif
+#endif // if defined(_MSC_FULL_VER) && _MSC_FULL_VER >= 193632528
 
 #elif _GTHREAD_USE_MUTEX_TIMEDLOCK || !defined(__unix__)
 using TimedMutex = std::timed_mutex;

--- a/include/fastrtps/utils/TimedMutex.hpp
+++ b/include/fastrtps/utils/TimedMutex.hpp
@@ -23,10 +23,16 @@
 #include <iostream>
 
 #if defined(_WIN32)
+
+#if defined(_MSC_FULL_VER) && _MSC_FULL_VER >= 193632528
+#include <mutex>
+#else
 #include <thread>
 extern int clock_gettime(
         int,
         struct timespec* tv);
+#endif
+
 #elif _GTHREAD_USE_MUTEX_TIMEDLOCK
 #include <mutex>
 #else
@@ -38,14 +44,12 @@ namespace fastrtps {
 
 #if defined(_WIN32)
 
+#if defined(_MSC_FULL_VER) && _MSC_FULL_VER >= 193632528
+using TimedMutex = std::timed_mutex;
+using RecursiveTimedMutex = std::recursive_timed_mutex;
+#else
 class TimedMutex
 {
-    // On MSVC 19.36.32528.95 `xtime` was changed into `_timespec64`.
-    // See https://github.com/eProsima/Fast-DDS/issues/3451
-    // See https://github.com/microsoft/STL/pull/3594
-#if defined(_MSC_FULL_VER) && _MSC_FULL_VER >= 193632528
-    using xtime = _timespec64;
-#endif  // _MSC_FULL_VER check
 
 public:
 
@@ -190,6 +194,8 @@ private:
 
     _Mtx_t mutex_;
 };
+#endif
+
 #elif _GTHREAD_USE_MUTEX_TIMEDLOCK || !defined(__unix__)
 using TimedMutex = std::timed_mutex;
 using RecursiveTimedMutex = std::recursive_timed_mutex;

